packages/react-components/react-tree/src/components/TreeItem/useTreeItem.ts

```diff
@@ -24,51 +20,51 @@ export function useTreeItem_unstable<Value = string>(
   props: TreeItemProps<Value>,
   ref: React.Ref<HTMLDivElement>,
 ): TreeItemState {
-  const [children, subtreeChildren] = React.Children.toArray(props.children);
-
   const contextLevel = useTreeContext_unstable(ctx => ctx.level);
 
   const id = useId('fui-TreeItem-', props.id);
 
   const {
-    content,
-    subtree,
-    expandIcon,
-    leaf: isLeaf = subtreeChildren === undefined,
-    actions,
-    as = 'div',
     onClick,
     onKeyDown,
-    ['aria-level']: level = contextLevel,
+    as = 'div',
     value = id,
+    itemType = 'leaf',
+    'aria-level': level = contextLevel,
     ...rest
   } = props;
 
   const requestOpenChange = useTreeContext_unstable(ctx => ctx.requestOpenChange);
   const requestNavigation = useTreeContext_unstable(ctx => ctx.requestNavigation);
 
-  const isBranch = !isLeaf;
+  const [isActionsVisible, setActionsVisible] = React.useState(false);
+  const [isAsideVisible, setAsideVisible] = React.useState(true);
+
+  const handleActionsRef = (actions: HTMLDivElement | null) => {
+    setAsideVisible(actions === null);
+  };
+
+  const isBranch = itemType === 'branch';
 
   const open = useTreeContext_unstable(ctx => isBranch && ctx.openItems.has(value));
-  const { dir, targetDocument } = useFluent_unstable();
-  const expandIconRotation = open ? 90 : dir !== 'rtl' ? 0 : 180;
 
-  const actionsRef = React.useRef<HTMLElement>(null);
-  const expandIconRef = React.useRef<HTMLElement>(null);
-  const subtreeRef = React.useRef<HTMLElement>(null);
+  const actionsRef = React.useRef<HTMLDivElement>(null);
+  const expandIconRef = React.useRef<HTMLDivElement>(null);
+  const layoutRef = React.useRef<HTMLDivElement>(null);
+  const subtreeRef = React.useRef<HTMLDivElement>(null);
 
   const handleArrowRight = (event: React.KeyboardEvent<HTMLDivElement>) => {
     if (!open && isBranch) {
```

Comment:
you could probably do an early return if the tree item isn't a branch

Reply:
I'll se to split into 2 hooks, one for the branch and one for the leaf as the `itemType` value should not change